diff --git a/acinclude.m4 b/acinclude.m4
index a581e1e..c43ffce 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -712,8 +712,8 @@ AC_DEFUN([OVS_CHECK_LINUX_COMPAT], [
                                    [int nf_conncount_add],
                                    [], [OVS_DEFINE([HAVE_UPSTREAM_NF_CONNCOUNT])])])
 
-  OVS_GREP_IFELSE([$KSRC/include/linux/random.h], [prandom_u32])
-  OVS_GREP_IFELSE([$KSRC/include/linux/random.h], [prandom_u32_max])
+  OVS_GREP_IFELSE([$KSRC/include/linux/prandom.h], [prandom_u32])
+  OVS_GREP_IFELSE([$KSRC/include/linux/prandom.h], [prandom_u32_max])
 
   OVS_GREP_IFELSE([$KSRC/include/net/rtnetlink.h], [get_link_net])
   OVS_GREP_IFELSE([$KSRC/include/net/rtnetlink.h], [name_assign_type])
